src/info/info.c

```diff
@@ -24,358 +24,412 @@ Info Handler
 };
 
-OBJECT_DEFINE_FREE(INFO);
+struct InfoSave
+{
+    MemContext *memContext;                                         // Mem context
+    IoWrite *write;                                                 // Write object
+    IoFilter *checksum;                                             // hash to generate file checksum
+    String *sectionLast;                                            // The last section seen
+};
+
+/***********************************************************************************************************************************
+Macros and buffer constants for checksum generation
+***********************************************************************************************************************************/
+#define INFO_CHECKSUM_BEGIN(checksum)                                                                                              \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        ioFilterProcessIn(checksum, BRACEL_BUF);                                                                                   \
+    }                                                                                                                              \
+    while (0)
+
+BUFFER_STRDEF_STATIC(INFO_CHECKSUM_SECTION_END_BUF, "\":{");
+
+#define INFO_CHECKSUM_SECTION(checksum, section)                                                                                   \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        ioFilterProcessIn(checksum, QUOTED_BUF);                                                                                   \
+        ioFilterProcessIn(checksum, BUFSTR(section));                                                                              \
+        ioFilterProcessIn(checksum, INFO_CHECKSUM_SECTION_END_BUF);                                                                \
+    }                                                                                                                              \
+    while (0)
+
+BUFFER_STRDEF_STATIC(INFO_CHECKSUM_SECTION_NEXT_END_BUF, "},");
+
+#define INFO_CHECKSUM_SECTION_NEXT(checksum)                                                                                       \
+    do                                                                                                                             \
+    {                                                                                                                              \
```

Comment:
This is to make the macro appear as a single statement just like a function.  Consider:
```
if (true)
    INFO_CHECKSUM_SECTION_NEXT(checksum);
else
    <something>
```
This will not compile if the macro is multi-line or just has braces around it because the `;` will end the `if` statement and the `else` will be orphaned.  You could leave off the `;` in these cases or be sure to wrap it in braces but that is hard to remember and the errors obscure.

Compilers know this trick and will remove the loop during optimization.